pilot/pkg/networking/core/v1alpha3/route/route.go

```diff
@@ -464,6 +466,23 @@ func translateRoute(
 	if in.CorsPolicy != nil {
 		out.TypedPerFilterConfig[wellknown.CORS] = protoconv.MessageToAny(TranslateCORSPolicy(in.CorsPolicy))
 	}
+	if len(hostnames) > 0 {
+		for _, hostname := range hostnames {
+			statefulSvc := serviceRegistry[hostname]
+			if statefulConfig := util.MaybeBuildStatefulSessionFilterConfig(statefulSvc); statefulConfig != nil {
+				if out.TypedPerFilterConfig == nil {
+					out.TypedPerFilterConfig = make(map[string]*anypb.Any)
+				}
```

Comment:
Most important issue (or readability problem) is what happens if more than one service has session affinity. Do we add 2 filters ?

Reply:
We add only one and break.  This only happens in weighted cluster mode (and most cases it applies weights on same destination(